client/app/pages/dashboards/hooks/useDashboard.js

```diff
@@ -63,15 +64,17 @@ function useDashboard(dashboardData) {
   const updateDashboard = useCallback(
     (data, includeVersion = true) => {
       setDashboard(currentDashboard => extend({}, currentDashboard, data));
-      // for some reason the request uses the id as slug
-      data = { ...data, slug: dashboard.id };
+      data = { ...data, id: dashboard.id };
       if (includeVersion) {
         data = { ...data, version: dashboard.version };
       }
       return Dashboard.save(data)
-        .then(updatedDashboard =>
-          setDashboard(currentDashboard => extend({}, currentDashboard, pick(updatedDashboard, keys(data))))
-        )
+        .then(updatedDashboard => {
+          setDashboard(currentDashboard => extend({}, currentDashboard, pick(updatedDashboard, keys(data))));
+          if (has(data, "name")) {
+            location.setPath(url.parse(updatedDashboard.url).pathname, true);
```

Comment:
You can use `navigateTo(..., true)`: second argument is to replace URL state - router will not reload controller in this case

Reply:
IIRC `navigateTo` replaces the url params, in this case I don't want them updated.

Reply:
Got it. Thanks for explanation!